group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupConfig.java

```
@@ -0,0 +1,111 @@
+
+/**
+ * Group configuration related parameters and supporting methods like validation, etc. are
+ * defined in this class.
+ */
+public class GroupConfig extends AbstractConfig {
+
+    public static final String CONSUMER_SESSION_TIMEOUT_MS_CONFIG = "consumer.session.timeout.ms";
+
+    public static final String CONSUMER_HEARTBEAT_INTERVAL_MS_CONFIG = "consumer.heartbeat.interval.ms";
+
+    private static final ConfigDef CONFIG = new ConfigDef()
+            .define(CONSUMER_SESSION_TIMEOUT_MS_CONFIG, INT, GroupCoordinatorConfig.CONSUMER_GROUP_SESSION_TIMEOUT_MS_DEFAULT, atLeast(1), MEDIUM, GroupCoordinatorConfig.CONSUMER_GROUP_SESSION_TIMEOUT_MS_DOC)
+            .define(CONSUMER_HEARTBEAT_INTERVAL_MS_CONFIG, INT, GroupCoordinatorConfig.CONSUMER_GROUP_HEARTBEAT_INTERVAL_MS_DEFAULT, atLeast(1), MEDIUM, GroupCoordinatorConfig.CONSUMER_GROUP_HEARTBEAT_INTERVAL_MS_DOC);
+    public GroupConfig(Map<?, ?> props) {
+        super(CONFIG, props, false);
+    }
+
+    public static Set<String> configNames() {
+        return new HashSet<>(CONFIG.names());
+    }
+
+    /**
+     * Check that property names are valid
+     */
+    public static void validateNames(Properties props) {
+        Set<String> names = configNames();
+        for (Object name : props.keySet()) {
+            if (!names.contains(name)) {
+                throw new InvalidConfigurationException("Unknown group config name: " + name);
+            }
+        }
+    }
+
+    /**
+     * Validates the values of the given properties.
+     */
+    public static void validateValues(Map<?, ?> valueMaps, Properties groupConfigBounds) {
+        int consumerHeartbeatInterval = (Integer) valueMaps.get(CONSUMER_HEARTBEAT_INTERVAL_MS_CONFIG);
+        int consumerSessionTimeout = (Integer) valueMaps.get(CONSUMER_SESSION_TIMEOUT_MS_CONFIG);
+        require(consumerHeartbeatInterval >= (int) groupConfigBounds.get(GroupCoordinatorConfig.CONSUMER_GROUP_MIN_HEARTBEAT_INTERVAL_MS_CONFIG),
```

Coming back to a previous point, I wonder if we should throw `InvalidConfigurationException` here.

---

Done.